### ISSUE TITLE
Appsody deploy delete should know what to delete

### ISSUE THREAD

**Issue by henrynash** [appsody/appsody#602]
Given that I assume our design criteria is that a given appsody project directory only has one deploy live at a time, running `appsody deploy delete` should always know how to delete the current deploy. Today this isn't true, e.g. if you do:

`appsody deploy -tag ABC --pull-url 123 --pull-url 987--namespace XYZ`

then an `appsody deploy delete` will fail unless you specify the namespace on the delete command. This isn't intuitive...and if my assumption at the start is true, then should not be necessary.

**Comment by neeraj-laad** [comment]
I agree we want to make it easy for users to delete what they deployed.

@chilanti might have some thoughts on this.

**Comment by chilanti** [comment]
We did think about this a bit way back when the function was implemented. It's possible to add the `namespace` to the deployment manifest as it is generated (`appsody deploy delete` simply runs `kubectl delete -f app-deploy.yaml`). So - if the `namespace` was part of the deployment manifest, we could default `appsody deploy delete` to that namespace.
However... remember that `appsody deploy` could be used multiple times to deploy to different namespaces. At that point, the user would still have to remember which namespace `delete` refers to - Appsody would remember just the last instance. @EnriqueL8 - @kylegc - since `app-deploy.yaml` is generated at build time, we'd have to discuss how to handle injecting the namespace into the manifest.

**Comment by henrynash** [comment]
@chilanti Ok, so that is good to know...which is why I started my description of the issue with "Given that I assume..." !! So my assumption was wrong - it is a design goal that you can deploy to multiple places from one appsody project directory (I had not seen this expressed anywhere in the docs).

Given that, please lets fix #464, since without adding the namespace to the app-config doesn't work unless you also use the --namespace option (which is the whole thing I am trying to allow people to avoid if they want delete to know what to do!).
